### PR TITLE
refactor: move /info input/ouput to oatpp::dto

### PR DESCRIPTION
**Pull request by sileht**

### CODE REVIEW

File: src/http/controller.hpp [hunk 1]
```diff
@@ -63,12 +67,37 @@ class DedeController : public oatpp::web::server::api::ApiController
            REQUEST(std::shared_ptr<IncomingRequest>, req),
            QUERIES(QueryParams, queryParams))
   {
-    // TODO(sileht): why do serialize the query string to char*
-    // to later get again a APIData...
+    auto jsonObjectMapper
```
Comment (lganzzzo) [inline src/http/controller.hpp:70]: You may create this object-mapper on the stack:

```cpp
oatpp::parser::json::mapping::ObjectMapper jsonObjectMapper;
```

Or, if the controller has it's own mapper - use that one:

```cpp
getDefaultObjectMapper();
```

File: src/http/controller.hpp [hunk 2]
```diff
@@ -63,12 +67,37 @@ class DedeController : public oatpp::web::server::api::ApiController
...
+        = oatpp::parser::json::mapping::ObjectMapper::createShared();
     auto req_start_time = std::chrono::steady_clock::now();
-    std::string jsonstr = _oja->uri_query_to_json(queryParams);
-    auto janswer = _oja->info(jsonstr);
-    return _oja->jdoc_to_response(req, janswer, req_start_time);
+    auto info_resp = InfoResponse::createShared();
+    info_resp->head = InfoHead::createShared();
+    info_resp->head->services
```
Comment (lganzzzo) [inline src/http/controller.hpp:75]: You can do simply the following:

```cpp
info_resp->head->services = {}; // initialize an empty (not null) list.
```

File: src/http/controller.hpp [hunk 3]
```diff
@@ -63,12 +67,37 @@ class DedeController : public oatpp::web::server::api::ApiController
...
+        auto json_str = _oja->jrender(jd);
+        auto service_info
+            = jsonObjectMapper->readFromString<oatpp::Object<Service>>(
+                json_str.c_str());
+        info_resp->head->services->emplace_back(service_info);
+        ++hit;
+      }
+
+    auto resp = createDtoResponse(Status::CODE_200, info_resp);
+    return _oja->finalize_response(req, resp, req_start_time, "");
```
Comment (lganzzzo) [inline src/http/controller.hpp:100]: Note: there are `ResponseInterceptors` available now - https://github.com/oatpp/oatpp/blob/master/changelog/1.2.5.md#introduce-responseinterceptor

Consider using a response interceptor instead.

File: src/oatppjsonapi.cc [hunk 4]
```diff
@@ -223,6 +223,33 @@ namespace dd
...
+  OatppJsonAPI::finalize_response(
+      const std::shared_ptr<
+          oatpp::web::server::api::ApiController::IncomingRequest> &request,
+      std::shared_ptr<oatpp::web::protocol::http::outgoing::Response>
+          &response,
+      const std::chrono::time_point<std::chrono::steady_clock> &req_start_time,
+      const std::string &service)
+  {
+    if (!FLAGS_allow_origin.empty())
+      response->putHeader("Access-Control-Allow-Origin",
```
Comment (lganzzzo) [inline src/oatppjsonapi.cc:249]: Just use the following:

```cpp
response->putHeader("Access-Control-Allow-Origin", FLAGS_allow_origin.c_str());
```

Also, note:
There is an out-of-the-box CORS support in oatpp:
- Per Endpoint with `ADD_CORS` macro
- And global - per connection handler - see [1.2.5 changelog](https://github.com/oatpp/oatpp/blob/master/changelog/1.2.5.md#enable-global-cors)

**Review by lganzzzo — COMMENTED** [review]
Hey guys,

Feel free to ping me if you need a code-review.

Regards,
Leonid

**Review by beniz — APPROVED** [review]

**Comment by sileht** [comment]
@Mergifyio rebase

**Comment by sileht** [comment]
@Mergifyio rebase

**Review by fantes — APPROVED** [review]

**Comment by sileht** [comment]
@Mergifyio rebase

**Pull request merged**